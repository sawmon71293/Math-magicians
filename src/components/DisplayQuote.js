// eslint-disable-next-line import/no-extraneous-dependencies
import axios from 'axios';
import { useState, useEffect } from 'react';


export default function DisplayQuote() {
  const apiUrl = 'https://api.api-ninjas.com/v1/quotes/';
  const token = 'CPEgr5h8YYAQbPBSC4RRnw==vFulvycB9TQ8Wyah';
  const [quote, setQuote] = useState('');
  const [author, setAuthor] = useState('');
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState('');

  useEffect(() => {
    axios({
      method: 'GET',
      url: `${apiUrl}`,
      headers: { 'X-API-Key': `${token}` },
    })
      .then((res) => {
        setAuthor(res.data[0].author);
        setQuote(res.data[0].quote);
        setIsLoading(false);
      })
      .catch((err) => {
        setError('An error occurred while fetching the quote.');
        setIsLoading(false);
        throw new Error(err);
      });
  }, []);

  return (
    <section className="container quote">
      <span>{isLoading ? 'Loading ...' : ''}</span>
      <span>{error || ''}</span>
      <span>{quote ? `${quote} _ ${author}` : ''}</span>
    </section>
  );
}
