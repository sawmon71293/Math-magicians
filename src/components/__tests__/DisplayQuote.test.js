import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import DisplayQuote from '../DisplayQuote';

describe('Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DisplayQuote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
