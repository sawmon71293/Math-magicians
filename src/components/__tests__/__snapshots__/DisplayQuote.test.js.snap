// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<section
  className="container quote"
>
  <span>
    Loading ...
  </span>
  <span />
  <span />
</section>
`;
